### PR TITLE
[2] User account registration

### PR DESCRIPTION
**Pull request by Riwuko**
[2] User account registration

*Description:* This branch adds new user registration module.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+rebuild:
+	@make clean
+	@make build
+
+
+
+# Set LOGS=true if you want run dev environment with outputting logs
+run:
+	docker-compose up `if ! [ -n "${LOGS}" ]; then echo "-d"; fi`
```
Comment (robertmusil) [inline Makefile]: hey @Riwuko can you please explain what does it do

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+
+# Run nosetests if you use run_
+#run_tests:
+#	docker exec web bash -c " "
```
Comment (robertmusil) [inline Makefile]: you really don't want to leave any commented out code ever.

File: config/settings.py [hunk 3]
```diff
@@ -28,51 +28,58 @@
...
 INSTALLED_APPS = [
-    'django.contrib.admin',
-    'django.contrib.auth',
-    'django.contrib.contenttypes',
-    'django.contrib.sessions',
-    'django.contrib.messages',
-    'django.contrib.staticfiles',
-
-    'kanban',
+    "user_account",
```
Comment (robertmusil) [inline config/settings.py:38]: next time, if you run a formatter on a whole code base, you ideally want to make a separate PR for that, because it makes it really hard to discern what was actually changed or not

File: user_account/api/serializers.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
+from rest_framework import serializers
+from user_account.models.user import UserAccount
+
+
+class RegistrationSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = UserAccount
+        fields = ["email", "username", "password"]
```
Comment (robertmusil) [inline user_account/api/serializers.py]: why "username", this field is not required in the spec

File: user_account/api/views.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+from rest_framework.decorators import api_view
+
+from user_account.api.serializers import RegistrationSerializer
+
+
+@api_view(
+    [
+        "POST",
+    ]
+)
```
Comment (robertmusil) [inline user_account/api/views.py]: this decorator looks like a formatting oopsie, why not in a single line?

File: user_account/models/account_manager.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
+from django.db import models
+from django.contrib.auth.models import BaseUserManager
+
+
+class AccountManager(BaseUserManager):
+    def create_user(self, email, username, password=None):
+        if not email:
+            raise ValueError("Users must have an email address")
+        if not username:
+            raise ValueError("Users must have a username")
```
Comment (robertmusil) [inline user_account/models/account_manager.py]: You shouldn't output errors one by one, ask Justyna for help here

File: user_account/api/views.py [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+def registration_view(request):
```
Comment (nitz14) [inline user_account/api/views.py]: Can you change this to generic Class view ?

**Review by robertmusil — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (Riwuko) [reply Makefile]: Oops, I thought I removed this, sorry.

File: user_account/models/account_manager.py [hunk 8]
```diff
@@ -0,0 +1,19 @@
+from django.db import models
+from django.contrib.auth.models import BaseUserManager
+
+
+class AccountManager(BaseUserManager):
```
Comment (nitz14) [inline user_account/models/account_manager.py:5]: what with create superuser ?

File: config/settings.py (hunk 3 above)
Comment (Riwuko) [reply config/settings.py:38]: I know, I didn't mean that, it was an accident, sorry. :(

File: user_account/api/views.py (hunk 5 above)
Comment (Riwuko) [reply user_account/api/views.py]: It was black who did it, how can I tell him not to next time?

File: user_account/api/serializers.py [hunk 9]
```diff
@@ -0,0 +1,16 @@
+from rest_framework import serializers
+from user_account.models.user import UserAccount
```
Comment (nitz14) [inline user_account/api/serializers.py:3]: use some tool to sort imports.

for example:
https://medium.com/@cereblanco/setup-black-and-isort-in-vscode-514804590bf9

isort

**Review by nitz14 — APPROVED** [review]
